### ISSUE TITLE
How to delete file in pastebin.com

### ISSUE THREAD

**Issue by xfsrp** [ShareX/ShareX#3502]
I uploaded a text file by accident, and it was uploaded to pastebin.com. How do I delete this pastebin.com URL in ShareX?

Print I accidentally clicked here: https://i.imgur.com/FDZHoGR.png

Can you help me delete it?

**Comment by lordgreggreg** [comment]
Once uploaded to pastebin, contacting them and reporting it is the only way to remove it before it expires.
[Contact Us - Pastebin](https://pastebin.com/contact)

**Comment by manvankaliber** [comment]
How about you guys don't add the submenu function to important files when not logged-in or disable it by default? 

I just uploaded a .PHP file and now I am scared for life that I actually upload something truly important like my bank info because you guys put the 'upload to pastebin' context menu entry right below the native windows 'open file' entry. 

It doesn't even ask for confirmation, it just straight up uploads it.

Please consider changing this?

**Comment by jhorv-bvb** [comment]
I second this because it just happened to me - I accidentally uploaded a random text file that luckily was nothing.  However, right next to it in Windows Explorer was a file containing a dump of internal network addresses in my company.  At the very least I think ShareX should default to showing a confirmation prompt.

**Comment by UltraCakeBakery** [comment]
@Jaex you mind re-opening this issue?

**Comment by Snigelson** [comment]
At the very least it should not be included in the context menu by default.

I have now uninstalled ShareX. Very bad first experience. 0/10

**Comment by penz6** [comment]
I think there definitely needs to be a first time upload confirmation for this.

**Comment by DasDunkel** [comment]
> 
> 
> I think there definitely needs to be a first time upload confirmation for this.

You see, we already do have a popup from the first upload, people were just clicking yes to it without even reading it so we've now made it even harder to do by adding a red banner and a timer on the yes button: https://dunk.dev/w9CyTH2d.png

**Comment by turchinc** [comment]
this feature is a serious security risk. you add a new menu item to my default context menu, and fucking upload data without confirmation to an anonymous fucking site that I cannot even fucking delete? what the fuck dude?

**Comment by turchinc** [comment]
and it is maybe one thing for an image, this feature has no reason to exist for any other file types. I use sharex for making screenshots and not for accidentally uploading source code to anonymous sites just because your menu entry is right next to the "diff this file" menu entry...

**Comment by Crumbles-Pieces** [comment]
Wow I'm uploading things I never intended to. Oops misclicking is consequential beyond imagination

**Comment by vinivosh** [comment]
Was this seriously never changed? Lmao

**Comment by breenbob** [comment]
This is downright dangerous and irresponsible. I also think this issue should be reopened @Jaex. Files are uploaded without prompt, anonymously, with no expiration, and no way for the user to delete other than by contacting pastebin and hoping you get a response. As pointed out the context menu option appears right at the top, making it super easy to click by accident.

For anyone else who wants to keep using ShareX (great application) without this major security risk, check out [this issue](https://github.com/ShareX/ShareX/issues/2162). I feel like this setting should be disabled by default...

**Comment by DasDunkel** [comment]
First off, it's not "right at the top", it's the third option in the list. And if you're talking about right click context menus in Explorer, you can disable the explorer integration at `Application Settings > Integration > Show "Upload with ShareX" button`

It's not our fault that you accidentally click things enough times to get annoyed by it, adding a popup/confirmation for each use would just make it annoying for everyone who uses this feature.

**Comment by UltraCakeBakery** [comment]
> First off, it's not "right at the top", it's the third option in the list. And if you're talking about right click context menus in Explorer, you can disable the explorer integration at `Application Settings > Integration > Show "Upload with ShareX" button`
> 
> It's not our fault that you accidentally click things enough times to get annoyed by it, adding a popup/confirmation for each use would just make it annoying for everyone who uses this feature.

It is stupid that it is enabled by default. The first time it happend to my collegae he uploaded his recovery codes for our deployment server. One time is more than enough to get annoyed by it. 

Adding a popup/confirmation is a great idea. For people like you they can add a "don't ask me again" checkbox. They should make that disabled by default so you can forget to click it. I mean, "It's not our fault that you accidentally click things".

You have a shitty fucking mentality kid. Grow up.

**Comment by turchinc** [comment]
erm, yes right at the top, next to diff... in the windows context menu:
![image](https://user-images.githubusercontent.com/645266/167680888-119b4a57-dbab-4b0d-a8f6-80592b3724eb.png)
and actually yes, it is the fault of the application for having an "active by default" philosophy for uploading user data to an anonymous site and then placing that in a prominent menu entry in the default windows context that is far too easy to click on accident. 

good to know now that hidden somewhere three levels down in a functionally overloaded context menu there is a setting, that if you know it exists,  can actually be disabled.

**Comment by CubityFirst** [comment]
When it comes down to it, the vast majority of people want to upload images, text, or videos with ShareX.

Why disable a feature that 99% of people would want enabled by default, or spam the user with annoying popups. We ensure that before you actually upload anything, that you want to upload. We lock the user out for 5 seconds from even pressing yes, and it has a bright red banner.

It's literally entirely your fault if you press yes to confirming that you want ShareX to have the ability to upload, and then press Upload with ShareX.

> "It is stupid that it is enabled by default. The first time it happend to my colleague he uploaded his recovery codes for our deployment server. One time is more than enough to get annoyed by it."
Oh no! I took a screenshot of my banking details, my medical records, my bitcoin and crypto wallets and then went out of my way to upload it to the public internet!

> "I mean, we can just apply your "It's not our fault that you accidentally click things" sentiment to the confirm button while we're at it..."
We literally lock the button for 5 seconds so you're unable to continue without it. If you press no (which is unlocked right away), you have to then go out of your way to re-enable uploading.

> "and actually yes, it is the fault of the application for having an "active by default" philosophy for uploading user data to an anonymous site and then placing that in a prominent menu entry in the default windows context that is far too easy to click on accident."
There's a lot of applications that assume user behaviour in one way or another. I don't see you complaining that Edit with Code or Edit with Notepad++ exist. I don't see you complaining about the fact that you don't have to go through a long setup process with every variable that could be set blanked and nulled out each time you install a program. People that develop programs have to take a little bit of initiative. If 99% of the users of an application want to use it a specific way, that specific way should probably be the default.

ShareX also doesn't have "active by default" for uploading, the first time you try to upload anything it literally asks you. You give your consent for things to be uploaded. It's entirely your fault if you then press the button to upload things.

TLDR: Why disable uploading for everyone by default, when most people want that to happen. We already ask you if you want to upload, you said yes, you then proceeded to upload something you didn't want to.

**Comment by CubityFirst** [comment]
I can't imagine the number of support requests, github issues, and other related posts that would happen if uploading was disabled by default and you had to go out of your way to enable uploading. Just invisioning the flood of "Hurr durr ShareX is bad and won't upload!!?!" is sickening. 

People don't complain that Gyazo, Lightshot, Greenshot etc, all have the ability to upload, and to my best understanding, don't warn the user first.

The only four ways that we could proceed from here are ;

Upload by Default with no warning.
As you said before, this isn't ideal, and could lead to even more people uploading things that they might not otherwise want. It does decrease the user traction though, and it's one less box that appears.

Ask for permission on every single upload.
This is something that is possible, but would increase friction to such a high degree that I feel like the adoption rate of the program would suffer, especially since most people probably want to take quick screenshots, and not get forced to click accept each and every time. If there was a "Don't ask again button", then wouldn't this cause exactly the same issue that you guys are having now? You'd say "Don't ask again", and then in 6 months time accidently upload your 2FA recovery codes, and then come blame us anyway.

Disable Upload by Default.
Most users want to upload their screenshots, so while it might save the very few people who are uploading sensitive information, it yet again forces hundreds of thousands of people into a situation where they have to go change a setting, instead of just a limited number disabling uploads voluntarily. 

Ask the user.
This is what we've settled on. We pop up a prompt, make sure that the user is aware of what's going on, lock the user out from pressing accept for a few seconds to give them time to read it (and if they spam whatever button you can click) it'll disable uploads. Most users are happy to have that one time prompt, so it has the nature of "Don't ask me again", and then they don't have to go out of their way to their settings to re-enable uploading. The people who know that they're going to either be working with sensitive documents, or know that they don't want to upload also have the option right then and there to disable it. If either party wishes to change their mind, they can go to their settings and do that.

**Comment by CubityFirst** [comment]
In regards to the windows context menu, We have the same few paths. Either show it by default, don't show it unless selected, or ask. Most people probably want it enabled, so it's enabled by default.

We could go down this rabbit hole with every single feature. Should we ask every user on if they want to install X specific browser extension so that they can upload with that? Most users probably don't care about this feature, so it's disabled by default, and left in a menu so that those who do use it, can go there, and renable it.

Should we quiz them on what their blood type is? Light Mode, or Dark Mode? What about X feature, or Y. It's untenable. 

I understand that a great deal of these three posts is rehashing the same point over and over.

Jaex has to make some base level assumptions about how people will use the program that he has made, noone wants to download an application for the first time and be greeted by a form that takes two hours to fill out. Sure, the rate of people cocking up after that is going to be lower, but it will also dissuade the vast majority of people from using the application. I don't think that it's "dangerous and irresponsible".

**Comment by CubityFirst** [comment]
In regards to the windows context menu, We have the same few paths. Either show it by default, don't show it unless selected, or ask. Most people probably want it enabled, so it's enabled by default.

We could go down this rabbit hole with every single feature. Should we ask every user on if they want to install X specific browser extension so that they can upload with that? Most users probably don't care about this feature, so it's disabled by default, and left in a menu so that those who do use it, can go there, and renable it.

Should we quiz them on what their blood type is? Light Mode, or Dark Mode? What about X feature, or Y. It's untenable. 

I understand that a great deal of these three posts is rehashing the same point over and over.

Jaex has to make some base level assumptions about how people will use the program that he has made, noone wants to download an application for the first time and be greeted by a form that takes two hours to fill out. Sure, the rate of people cocking up after that is going to be lower, but it will also dissuade the vast majority of people from using the application. I don't think that it's "dangerous and irresponsible".

**Comment by randomguyingithub** [comment]
up/confirmation is a great idea. For people like you they can add a "don't ask me again" checkbox. They should make that one disabled by default so you can forget to check it when you're in a hurry. I mean, we can just apply your "It's not our fault that you accidentally click things" sentiment to the `confirm` button while we're at it...
> 
> You have a shitty fucking mentality kid. Grow up.

Did you guys just read the title and doesn't read the reply and ironically said someone else mentality kid? Seriously? IT IS NOT enabled by default. There is a pop up ask you whether want to upload by default and you press ok and the setting is saved to a setting file. Either sysadmin somehow share the setting file to all computers and assume you're talking in office environment or you just press ok without reading anything.

**Comment by UltraCakeBakery** [comment]
Stop saying `most people` or throw around giant percentages without providing actual survey results or something similar.

You guys also collectively spend more time repeating each others previous statements than you would every spend on clicking 'yes' or quickly hitting your enter key in the confirmation modal.

Even though there now apparently is a confirmation modal, the `upload` contextmenu item is still positioned awfully. I would not blame anyone for clicking it on accident. It is as if your bank has a `donate all money to some random prince in Uganda` button right next to the actual `confirm transaction` button. If you are in a hurry, it is easy to make mistakes.

**Comment by UltraCakeBakery** [comment]
> But of course it spawns another similar problem: Should the context menu button be moved down by default?
If you can't guarantee where your UI elements show up, they shouldn't be included or enabled by default.

**Comment by CubityFirst** [comment]
> > But of course it spawns another similar problem: Should the context menu button be moved down by default?
> 
> If you can't guarantee where your UI elements show up, they shouldn't be included or enabled by default.

Hmmm.
![image](https://user-images.githubusercontent.com/14070268/167930666-b5fea83f-68bc-42eb-8e46-a49c6d259559.png)

There's a lot of applications that either add things to the context menu. Not just ShareX.

**Comment by UltraCakeBakery** [comment]
> > > But of course it spawns another similar problem: Should the context menu button be moved down by default?
> > 
> > 
> > If you can't guarantee where your UI elements show up, they shouldn't be included or enabled by default.
> 
> Hmmm. ![image](https://user-images.githubusercontent.com/14070268/167930666-b5fea83f-68bc-42eb-8e46-a49c6d259559.png)
> 
> There's a lot of applications that either add things to the context menu. Not just ShareX.

OK

**Comment by CubityFirst** [comment]
> Stop saying `most people` and throwing around giant percentages without providing actual survey results or something similar.
> 
> You guys also collectively spend more time repeating each others previous statements than you would every spend on clicking 'yes' or quickly hitting your enter key in the confirmation modal.
> 
> Even though there now apparently is a confirmation modal, the `upload` contextmenu item is still positioned awfully. I would not blame anyone for clicking it on accident. It is as if your bank has a `donate all money to some random prince in Uganda` button right next to the actual `confirm transaction` button. If you are in a hurry, it is easy to make mistakes.

Something that we forgot to mention before, we literally ask the user right then and there, on setup, if they want the context menu entry enabled. Sounds like you just spam clicked through another box.
![image](https://user-images.githubusercontent.com/14070268/167936688-3b66b8ee-301c-4d16-a042-1643ec4be9f1.png)

**Comment by UltraCakeBakery** [comment]
> > Stop saying `most people` and throwing around giant percentages without providing actual survey results or something similar.
> > You guys also collectively spend more time repeating each others previous statements than you would every spend on clicking 'yes' or quickly hitting your enter key in the confirmation modal.
> > Even though there now apparently is a confirmation modal, the `upload` contextmenu item is still positioned awfully. I would not blame anyone for clicking it on accident. It is as if your bank has a `donate all money to some random prince in Uganda` button right next to the actual `confirm transaction` button. If you are in a hurry, it is easy to make mistakes.
> 
> Something that we forgot to mention before, we literally ask the user right then and there, on setup, if they want the context menu entry enabled. Sounds like you just spam clicked through another box. ![image](https://user-images.githubusercontent.com/14070268/167936688-3b66b8ee-301c-4d16-a042-1643ec4be9f1.png)

My original and current complains are that there is/was (whatever the fuck it is according to you all) no confirmation modal when clicked, and the horrible position of the item in the contextmenu. Not that the item is actually in the contextmenu.

**Comment by 20maty** [comment]
Literally happened to me twice in the last few days. The button is literally next to "Open with Code". Nothing wild, but I can't imagine the damage if you click this by accident on a cloned GitHub repository of a classified project with potentially thousands of files that are now on the public internet. And Pastebin is sometimes also indexed by Google search.

I dont have the time to read trough th thread, maybe it ha been already posted but heres a small tutorial on how to disable the context menu element without reinsalling ShareX

1. Open ShareX
2. Click  "Application settings" in the sidebar
3. Click "Integration" (3rd Item)
4. Disable the checkbox for the conext menu

**Comment by yuis-ice** [comment]
LMAO This is the worst feature ever.

**Comment by CubityFirst** [comment]
ShareX does warn you before uploading any data in the first instance.

**Comment by yuis-ice** [comment]
I **love** the ShareX project but the truth is that I had to spend time requesting deletion of the personal information uploaded to the Pastebin admins. They deleted for me that time fortunately, but once uploaded thing is irreversible. 

In my case I usually upload my *image files* to *my own FTP server*. Since I've been using ShareX for 6 or 7 years or so, the very first time I informed so is very long time ago, and it's for the image I tried to upload. And the time I accidentally uploaded a *txt file* through the context menu, yes It's the first time uploading of 1. the file format 2. Pastebin. (here's the point that the project can improve I think)

I wouldn't make the same mistake again at least for the next years, but for the unaware people I would leave this comment.
